server/py/framework/utils/clients/base_client.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import contextlib
+import copy
+import typing
+from abc import ABC, abstractmethod
+
+import aiohttp
+import fastapi
+from fastapi.concurrency import run_in_threadpool
+
+import mlrun.common.schemas
+import mlrun.errors
+import mlrun.utils.singleton
+from mlrun.utils import logger
+
+
+class BaseClient(ABC, metaclass=mlrun.utils.singleton.AbstractSingleton):
```

Comment:
We have other clients unrelated to Iguazio that do not share this bas client. I'd say this is more if a `BaseIguazioClient` (same for the file name).
You might even want to have a `clients/iguazio` module for all 3 files